### ISSUE TITLE
Failing Windows CI Test test_rq_10

### ISSUE THREAD

**Issue by geographika** [MapServer/MapServer#6987]
One of the Python tests has started failing on the Windows Appveyor builds: https://ci.appveyor.com/project/MapServer/mapserver/builds/48782653/job/pngvcla25g5g03gu#L601

This is unrelated to the most recent check-in, and is most likely caused by updates to the GISInternals SDK kit.
Most recent commits: https://github.com/gisinternals/buildsystem/commits/main/
This included a point update to GDAL - with changes listed at: https://github.com/OSGeo/gdal/compare/v3.7.2...v3.7.3

The test itself `test_rq_10` includes the text:

```
    # Scan results.  This query is for a transparent pixel within the "x" of
    # the cubewerx logo.  In the future the raster query may well stop returning
    # "offsite" pixels and we will need to update this test.
```

The test does however appear to run successfully on the GHA and Travis builds.

**Comment by geographika** [comment]
Update after debugging:

Issue is opening the source .gif at the line below:

https://github.com/MapServer/MapServer/blob/be1a370e6e30b12ca7f1e272d3ed1e50afebc062/src/maprasterquery.c#L782

`msRasterQueryByRect(): Image handling error. Unable to open file data\grid.gif for layer grid1 ... fatal error.`

However opening the gif is fine when using `map2img` which also uses `GDALOpenEx`. The difference is it passes in a list of allowed GDAL drivers `papszAllowedDrivers`. However if this is NULL all drivers should be accepted including the GIF one. 

https://github.com/MapServer/MapServer/blob/be1a370e6e30b12ca7f1e272d3ed1e50afebc062/src/mapraster.c#L649

**Comment by rouault** [comment]
> The difference is it passes in a list of allowed GDAL drivers `papszAllowedDrivers`

Hum weird . @geographika  What is the value of papszAllowedDrivers and *papszAllowedDrivers ?   They should only be non-NULL if there is a CONNECTIONOPTIONS block in the mapfile

**Comment by geographika** [comment]
> Hum weird . @geographika  What is the value of papszAllowedDrivers and *papszAllowedDrivers ?   They should only be non-NULL if there is a CONNECTIONOPTIONS block in the mapfile

Also NULL, so even though the calling code is different, the actual call parameters seem identical. 

I can recreate locally on Windows with the latest SDK, and it runs fine with the older SDK, so this appears to rule out MapServer code changes being responsible.
However valid images are created with map2img with the same Mapfile and new SDK...

**Comment by rouault** [comment]
hum, actually when looking at the logs of the failing test in https://ci.appveyor.com/project/MapServer/mapserver/builds/48782653/job/pngvcla25g5g03gu, one sees """"
-- Found GDAL: C:/projects/mapserver/sdk/release-1928-x64/lib/gdal_i.lib  
-- Found GDAL 3.9
"""

whereas in another pull request (https://github.com/MapServer/MapServer/pull/6988) that has just run, the test succeeds https://ci.appveyor.com/project/MapServer/mapserver/builds/48786546/job/a5ad3ir6w8lx9yl0 and the logs show """"
-- Found GDAL: C:/projects/mapserver/sdk/release-1928-x64/lib/gdal_i.lib  
-- Found GDAL 3.8
""""
So the failure would be related to the SDK having used for some time GDAL 3.9 (dev), and now it uses 3.8...

**Comment by geographika** [comment]
> So the failure would be related to the SDK having used for some time GDAL 3.9 (dev), and now it uses 3.8...

Good catch. The SDK yesterday was using GDAL 3.9. It was updated a couple of times since, and now uses GDAL 3.8. The CI tests are now working again (and I reran the failing ones from yesterday and they are also now passing). 
I'm unsure if there will be an issue moving to GDAL 3.9 or if this was some clash of dependencies in the SDK. 

It might be good to take a copy of the SDK and put on an OSGeo server to avoid these types of CI failures in future. We can then plan an update to a new SDK and address issues when ready.
